gobblin-runtime/src/main/java/gobblin/runtime/api/JobExecutionLauncher.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.api;
+
+import gobblin.annotation.Alpha;
+
+/**
+ * Placeholder
+ */
+@Alpha
+public interface JobExecutionLauncher {
```

Comment:
Actually, it will replace to JobLauncherFactory but the interface will use higher level of abstractions. It will be in terms of JobExecutions rather than Properties. The lifecycle of a job goes through JobSpec -> JobSchedule -> JobExecution and the interfaces should reflect that.